File: app/src/main/java/com/xmx/tango/common/log/OperationLogEntityManager.java
package com.xmx.tango.common.log;

import com.xmx.tango.common.data.sql.BaseSQLEntityManager;

import org.greenrobot.eventbus.EventBus;

import java.util.Date;

/**
 * Created by The_onE on 2016/9/4.
 */
public class OperationLogEntityManager extends BaseSQLEntityManager<OperationLog> {
    private static OperationLogEntityManager instance;

    public synchronized static OperationLogEntityManager getInstance() {
        if (null == instance) {
            instance = new OperationLogEntityManager();
        }
        return instance;
    }

    private OperationLogEntityManager() {
        tableName = "OperationLog";
        entityTemplate = new OperationLog();
        openDatabase();
    }

    public void addLog(String operation) {
        OperationLog entity = new OperationLog();
        entity.mOperation = operation;
        entity.mTime = new Date();
        insertData(entity);
        EventBus.getDefault().post(new LogChangeEvent());
    }
}

File: app/src/main/java/com/xmx/tango/common/log/OperationLogEntityManager.kt
package com.xmx.tango.common.log

import com.xmx.tango.common.data.sql.BaseSQLEntityManager

import org.greenrobot.eventbus.EventBus

import java.util.Date

/**
 * Created by The_onE on 2016/9/4.
 * 操作日志实体管理器
 */
object OperationLogEntityManager : BaseSQLEntityManager<OperationLog>() {

    init {
        tableName = "OperationLog"
        entityTemplate = OperationLog()
        openDatabase(null)
    }

    /**
     * 添加日志
     * @param operation 操作信息
     */
    fun addLog(operation: String) {
        val entity = OperationLog()
        entity.mOperation = operation
        entity.mTime = Date()
        insertData(entity)
        EventBus.getDefault().post(LogChangeEvent())
    }
}
